src/finalizer.rs

```diff
@@ -0,0 +1,113 @@
+use crate::error::Error;
+use kube::api::{Meta, PatchParams, PatchStrategy};
+use kube::{Api, Client};
+use serde_json::json;
+use serde::de::DeserializeOwned;
+
+
+/// Checks whether our own finalizer is in the list of finalizers for the provided object.
+pub fn has_finalizer<T>(resource: &T, finalizer: &str) -> bool
+    where T: Meta
+{
+    return match resource.meta().finalizers.as_ref() {
+        Some(finalizers) => finalizers.contains(&finalizer.to_string()),
+        None => false,
+    };
+}
+
+/// Adds our finalizer to the list of finalizers.
+pub async fn add_finalizer<T>(
+    client: Client,
+    resource: &T,
+    finalizer: &str
+) -> Result<T, Error>
+    where T: k8s_openapi::Resource + Clone + Meta + DeserializeOwned
+{
+    let name = &Meta::name(resource);
+    let namespace = &Meta::namespace(resource).expect("Resource is namespaced");
```

Comment:
Is this error message correct? To me this is triggered, when the namespace option is None i.e. the resource doesn't have a namespace..
Also, I am hesitant to panic in libraries, shouldn't we let the binary decide how to handle this error?